src/TokensTracker/Trackers/NEP-11/Nep11Tracker.cs

```diff
@@ -244,16 +244,32 @@ public JObject GetNep11Balances(JArray _params)
             }
             foreach (var key in map.Keys)
             {
-                balances.Add(new JObject
+                try
                 {
-                    ["assethash"] = key.ToString(),
-                    ["tokens"] = new JArray(map[key].Select(v => new JObject
+                    using var script = new ScriptBuilder();
+                    script.EmitDynamicCall(key, "decimals");
+                    script.EmitDynamicCall(key, "symbol");
+
+                    var engine = ApplicationEngine.Run(script.ToArray(), _neoSystem.StoreView, settings: _neoSystem.Settings);
+                    var symbol = engine.ResultStack.Pop().GetString();
+                    var decimals = engine.ResultStack.Pop().GetInteger();
+                    var name = NativeContract.ContractManagement.GetContract(_neoSystem.StoreView, key).Manifest.Name;
+
+                    balances.Add(new JObject
                     {
-                        ["tokenid"] = v.tokenid,
-                        ["amount"] = v.amount,
-                        ["lastupdatedblock"] = v.height
-                    })),
-                });
+                        ["assethash"] = key.ToString(),
+                        ["tokens"] = new JArray(map[key].Select(v => new JObject
+                        {
+                            ["tokenid"] = v.tokenid,
+                            ["name"] = name,
+                            ["symbol"] = symbol,
+                            ["decimals"] = decimals.ToString(),
```

Comment:
Why do we duplicate this data for every token? Shouldn't it be present at the same level as `assethash` since it's essentially an `assethash` contract's metadata?

Reply:
Yes, my bad